modules/discretedata/dataset.h

```diff
@@ -0,0 +1,209 @@
+    BaseChannelMap;
+
+class IVW_MODULE_DISCRETEDATA_API DataChannelMap {
+public:
+    DataChannelMap() = default;
+    virtual ~DataChannelMap() = default;
+
+    /** Default copy shares Channels */
+    DataChannelMap(const DataChannelMap& copy) = default;
+
+    // Methods
+public:
+    /** Add a new channel to the set
+    *   @param channel Pointer to data, takes memory ownership
+    *   @return Shared pointer for further handling
+    */
+    SharedChannel addChannel(Channel* channel);
+
+    /** Add a new channel to the set
+    *   @param channel Shared pointer to data, remains valid
+    */
+    void addChannel(SharedConstChannel channel);
+
+    /** Returns the first channel from an unordered list.
+    */
+    SharedConstChannel getFirstChannel() const;
+
+    /** Returns the first channel from an unordered list.
+    */
+    template <typename T>
+    std::shared_ptr<const DataChannel<T>> getFirstChannel() const;
+
+    /** Returns the specified channel, returns first instance found
+    *   @param name Unique name of requested channel
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    SharedConstChannel getChannel(const std::string& name,
+                                  GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Returns the specified channel if it is in the desired format, returns first instance found
+    *   @param name Unique name of requested channel
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    template <typename T>
+    std::shared_ptr<const DataChannel<T>> getChannel(
+        const std::string& name, GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Returns the specified channel if it is in the desired format, returns first instance found
+    *   @param name Unique name of requested channel
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    template <typename T>
+    std::shared_ptr<const DataChannel<T>> getChannel(
+        const char* name, GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Returns the specified buffer, converts to buffer or copies
+    *   @param name Unique name of requested buffer
+    *   @param definedOn GridPrimitive type the channel is defined on, default 0D vertices
+    */
+    template <typename T>
+    std::shared_ptr<const BufferChannel<T>> getAsBuffer(
+        const std::string& name, GridPrimitive definedOn = GridPrimitive::Vertex) const;
+
+    /** Remove channel from set by shared pointer, data remains valid if shared outside
+    *   Swaps position in vector
+    *   @param channel Shared pointer to data
+    *   @return Successfull - channel was saved in the set indeed
+    */
+    bool removeChannel(SharedConstChannel channel);
+
+    /** Number of channels currently held
+    */
+    ind numChannels() const { return ChannelSet.size(); }
+
```

Comment:
can I ask for all the existing names?

Reply:
Currently the map is public anyways. But a list of available channels would be a better solution, I agree.

Reply:
Added in e9b5b392941ed6bef3a2c9ed9c938f68114dae9a

Reply:
Added in e9b5b392941ed6bef3a2c9ed9c938f68114dae9a